### DISCUSSION TITLE
SDEverywhere without a mac

### DISCUSSION THREAD

**Discussion by DenoBeno** [climateinteractive/SDEverywhere#316] [Q&A] [answered]
I see that SDEverywhere is designed to work on a mac. Unfortunately, I don't have a mac PC and buying one just try out SDEverywhere is a bit of an overkill.

Does anyone know how to get SDEverywhere working without a mac? Can it somehow be made to work under Linux or windows? Is it possible to install a macOS in a virtual machine and use SDEverywhere that way?

**Comment by chrispcampbell** [comment, marked as answer]
Hi @DenoBeno, SDEverywhere works well  on Windows and Linux in addition to macOS.  If the current docs gave the impression that it’s for macOS only, note that I’m in the middle of rewriting the README to be more up to date with the latest project structure.  You can see the latest work in progress (and Quick Start) instructions here:
https://github.com/climateinteractive/SDEverywhere/tree/chris/265-readme#repository-structure

We use SDE on all three operating systems at Climate Interactive.

Let us know if you have questions as you get started.

**Reply by DenoBeno** [reply]
Hi, @chrispcampbell . I took a quick look at your updated documentation and it still contains:

**[Requirements](https://github.com/climateinteractive/SDEverywhere/tree/chris/265-readme#requirements)**
Using SDEverywhere requires the macOS operating system and the free [Xcode](https://itunes.apple.com/us/app/xcode/id497799835) development tools from Apple.



Could you please update that sentence ASAP?

This is the first thing people read and it will make them believe that SDEverwhere works only on Mac even if there are installation instructions for Linux & windows available further down the road. This same sentence is also in README file on top of the SDEverywhere repository, and simply changing it to "Initially SDEverywhere required Mac but it works on other OSs now, parts of documentation still haven't been updated to reflect this" (or such) would do the trick...

**Reply by DenoBeno** [reply]
Mac requirement is also in this document:

[Using SDEverywhere to Make a Vensim Model into a Web Application
](https://github.com/climateinteractive/SDEverywhere/blob/chris/265-readme/notes/Using%20SDEverywhere%20to%20Make%20a%20Vensim%20Model%20into%20a%20Web%20Application.md)
Revised: 2021-10-21

This tutorial shows you how to take your Vensim model and turn it into an interactive web application using the open-source SDEverywhere toolkit. **SDEverywhere currently requires the macOS operating system in development**, but the resulting web app can be run on any web server, or even offline using the Firefox browser.

Another thing that would be helpful is a link to "SDEverywhere Web App Configuration Reference document". Google search for this ref. document doesn't find anything and it's not in models/sir/model directory:

[Designing your web app](https://github.com/climateinteractive/SDEverywhere/blob/chris/265-readme/notes/Using%20SDEverywhere%20to%20Make%20a%20Vensim%20Model%20into%20a%20Web%20Application.md#designing-your-web-app)
SDEverywhere generates a web application based on a standard template merged with app configuration in CSV files. **Refer to the SDEverywhere Web App Configuration Reference document** for details. There is a fully worked-out example in the models/sir/model directory.

**Reply by chrispcampbell** [reply]
Hi @DenoBeno, yes, as mentioned the branch is in an incomplete state (outdated sections have not yet been removed).  I am on holiday this week and will reply here next week once I have finished merging my complete set of changes to the documentation.  In particular the “into a web app” doc will be getting an overhaul to show how to use the new plugin based architecture.  In the meantime, following the Quick Start instructions with e.g. the ‘examples/sir’ directory may help get you familiar with the process until the docs are more fully revised.
